File: vam/vam/src/it/us/web/action/vam/cc/trasferimenti/Add.java
/*
Copyright (C) AGPL-3.0  

This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.*/
package it.us.web.action.vam.cc.trasferimenti;

import it.us.web.action.GenericAction;
import it.us.web.bean.BGuiView;
import it.us.web.bean.vam.CartellaClinica;
import it.us.web.bean.vam.Clinica;
import it.us.web.bean.vam.Trasferimento;
import it.us.web.bean.vam.lookup.LookupOperazioniAccettazione;
import it.us.web.dao.GuiViewDAO;
import it.us.web.exceptions.AuthorizationException;
import it.us.web.exceptions.ValidationBeanException;

import java.util.Date;
import java.util.Set;

import org.apache.commons.beanutils.BeanUtils;

public class Add extends GenericAction {

	
	public void can() throws Exception
	{
		BGuiView gui = GuiViewDAO.getView( "CC", "EDIT", "MAIN" );
		can( gui, "w" );
		canCc(cc);
	}
	
	@Override
	public void setSegnalibroDocumentazione()
	{
		setSegnalibroDocumentazione("trasferimenti");
	}

	@SuppressWarnings("unchecked")
	public void execute() throws Exception
	{
		Trasferimento trasferimento = new Trasferimento();
		
		BeanUtils.populate( trasferimento, req.getParameterMap() );
		
		Boolean urgenza = booleanoFromRequest( "urgenza" );
			
		trasferimento.setCartellaClinica( cc );
		trasferimento.setOperazioniRichieste( (Set<LookupOperazioniAccettazione>) objectList( LookupOperazioniAccettazione.class, "op_" ) );

		
		trasferimento.setModified( new Date() );
		trasferimento.setModifiedBy( utente );
		trasferimento.setEntered( trasferimento.getModified() );
		trasferimento.setEnteredBy( utente );
		
		trasferimento.setClinicaOrigine( utente.getClinica() );
		trasferimento.setUrgenza( urgenza );
		
		//if( urgenza ) ora anche se non c'? urgenza si permette di scegliere la clinica di destinazione
		{
			Clinica cd = (Clinica) persistence.find( Clinica.class, interoFromRequest( "clinicaDestinazioneId" ) );
			trasferimento.setClinicaDestinazione( cd );
		}
		
		//La cartella clinica attuale viene chiusa dopo il trasferimento dell'animale
		cc.setDataChiusura(dataFromRequest("dataRichiesta"));
		
		
		
		validaBean( trasferimento,  new ToAdd()  );
		
		persistence.insert( trasferimento );
		cc.setModified( new Date() );
		cc.setModifiedBy( utente );
		
		cc.setDimissioniEntered(new Date());
		cc.setDimissioniEnteredBy(utente);
		
		persistence.update( cc );		
		persistence.commit();
		
		setMessaggio( "Richiesta di trasferimento eseguita con successo." );
		redirectTo( "vam.cc.trasferimenti.List.us" );
	}
}

File: vam/vam/WebContent/js/vam/richiesteIstopatologici/add.js
/*
Copyright (C) AGPL-3.0  

This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.*/
function checkform(form)
{
	var ret = true;
	
	if(isFuturo(form.dataRichiesta.value))
	{
		alert("Non è possibile inserire una DATA RICHIESTA futura");
		ret = false;
	}
	
	if(!controllaDataAnnoCorrente(form.dataRichiesta, 'Data richiesta'))
	{
		ret = false;
	}
	
	//Flusso 170 si commenta
	/*if(form.dataSmaltimento.value!='' && confrontaDate(form.dataRichiesta.value,form.dataSmaltimento.value)>0)
	{
		alert("La data dell'esame istopatologico non può essere successiva alla data dello smaltimento("+form.dataSmaltimento.value+")");	
		form.dataRichiesta.focus();
		return false;
	}*/
	
	
	if( form.idTumoriPrecedenti.value == 2 )//tumori precedenti = si
	{
		if( ret && !isVoidOrIntPositivo(form.t.value,'T',form.t) )
		{
			ret = false;
		}
		
		if( ret  && !isVoidOrIntPositivo(form.n.value,'N',form.n) )
		{
			ret = false;
		}
		
		if( ret && !isVoidOrIntPositivo(form.m.value,'M',form.m) )
		{
			ret = false;
		}
	}
	

	if( ret && !isVoidOrIntPositivo(form.dimensione.value,'Dimensione',form.dimensione) )
	{
		ret = false;
	}

	if(ret && form.lookupAutopsiaSalaSettoria.value == '')
	{
		alert("Inserire il laboratorio di destinazione al quale inviare la richiesta");			
		ret = false;
	}
	
	if(ret && form.padreSedeLesione.value == '-1')
	{
		alert("Inserire la sede della lesione");			
		ret = false;
	}
	
	if( ret )
	{
		attendere();
	}
	
	return ret;
}


function checkformIzsm(form)
{
	var ret = true;
	
	if(form.dataEsito.value=='')
	{
		alert("Inserire la data esito");
		form.dataEsito.focus();
		ret= false;
	}else if(isFuturo(form.dataEsito.value))
	{
		alert("Non e' possibile inserire una data esito futura");
		ret= false;
	}
	
	if( ret )
	{
		attendere();
	}
	
	return ret;
}
